catboost/private/libs/target/data_providers.cpp

```diff
@@ -531,6 +539,12 @@ namespace NCB {
                 rawData.GetTargetDimension() <= 1,
                 "Currently only multi-regression and survival objectives work with multidimensional target"
             );
+            
```

Comment:
I think this check will be redundant after adding of checks to `CheckPreprocessedTarget`

Reply:
probably. Do you want to maintain or remove it? I mean, the time lost on this check should not be a limit factor on the overal catboost algorith.

Reply:
ok.
Done